### PR TITLE
Create connection file in jupyter runtime dir

### LINKED ISSUE #8827: kernel connection file created in `/tmp/` rather than `~/.local/share/jupyter/runtime/`
## Environment data

-   VS Code version: 1.63.2
-   Jupyter Extension version (available under the Extensions sidebar): v2021.11.1001550889
-   Python Extension version (available under the Extensions sidebar): v2021.12.1559732655
-   OS (Windows | Mac | Linux distro) and version: Gentoo Linux / CentOS 7 / WSL Ubuntu 20.04
-   Python and/or Anaconda version: Python 3.8.5
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): N/A
-   Jupyter server running: Remote / WSL

## Expected behaviour

runtime kernel connection json file created in `~/.local/share/jupyter/runtime/` so that it can be recognized by `jupyter console --existing`

## Actual behaviour

runtime kernel connection json file created in `/tmp/` and `jupyter console --existing` shows error message
`[ZMQTerminalIPythonApp] CRITICAL | Could not find existing kernel connection file kernel*.json`

this is not a big problem for ipykernel since I can connect a jupyter console to the kernel according to the `%connect_info` message.
but it becomes a problem when I use irkernel in which the `%connect_info` method haven't been implemented.

## Steps to reproduce:

1. create an interactive window in VS Code.
2. run `%connect_info`
3. see `$> jupyter <app> --existing /tmp/tmp-1033rJp0TWSU8vOv.json` in cell output, indicating that the connection json file is created in `/tmp/`
4. run `jupyter console --existing`
5. get error message `[ZMQTerminalIPythonApp] CRITICAL | Could not find existing kernel connection file kernel*.json`

## Logs

<details>

<summary>Output for <code>Jupyter</code> in the <code>Output</code> panel 
</summary>

<p>

```
%connect_info
```

```
{
  "shell_port": 9002,
  "iopub_port": 9004,
  "stdin_port": 9003,
  "control_port": 9001,
  "hb_port": 9000,
  "ip": "127.0.0.1",
  "key": "0b43f78a-4183-4728-b3c6-d734dfa4fb12",
  "transport": "tcp",
  "signature_scheme": "hmac-sha256",
  "kernel_name": ""
}

Paste the above JSON into a file, and connect with:
    $> jupyter <app> --existing <file>
or, if you are local, you can connect with just:
    $> jupyter <app> --existing /tmp/tmp-1033rJp0TWSU8vOv.json
or even just:
    $> jupyter <app> --existing
if this is the most recent Jupyter kernel you have started.
```

</p>
</details>

### PR DESCRIPTION
**Pull request by DonJayamanne**
Fixes #8827

### CODE REVIEW

**Review by rchiodo — APPROVED** [review]

File: src/kernels/raw/finder/jupyterPaths.node.ts [hunk 1]
```diff
@@ -16,10 +17,20 @@ import {
 import { tryGetRealPath } from '../../../platform/common/utils.node';
 import { IEnvironmentVariablesProvider } from '../../../platform/common/variables/types';
 import { traceDecorators } from '../../../platform/logging';
+// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
+const untildify = require('untildify');
+
+const homeDir = untildify('~');
```
Comment (IanMatthewHuff) [inline src/kernels/raw/finder/jupyterPaths.node.ts:23]: I think we already have this in PathUtils.home

**Review by IanMatthewHuff — APPROVED** [review]
I'd probably use PathUtils to get home, but rest looks good.

File: src/kernels/raw/finder/jupyterPaths.node.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/kernels/raw/finder/jupyterPaths.node.ts:23]: yes, i felt it was unnecessary to add yet another object via DI when it's all the same.

Comment (DonJayamanne) [reply src/kernels/raw/finder/jupyterPaths.node.ts]: Nope, convinced it not the right thing to do.
Changed as requested.

**Pull request merged**